### DISCUSSION TITLE
TOC component error

### DISCUSSION THREAD

**Discussion by AngeloCG97** [facebook/docusaurus#6151] [Q&A] [answered]
I am using the TOC component, but it throws me this error

``` Cannot read properties of null (reading 'clientHeight') at getNavbarHeight ```

This is the way I use it and import it

![image](https://user-images.githubusercontent.com/51149817/146991529-bbe1aa24-0a92-4f48-83ea-c5b065990be4.png)
``` import TOC from '@theme/TOC' ```

This is the content that I am sending you within the toc parameter of the component

![image](https://user-images.githubusercontent.com/51149817/146991775-22e8fe50-d9d1-4ceb-a1c2-5a21d71a6d7b.png)

The error appears after I update to the latest version of docusaurus, before doing the update this worked without problem
![image](https://user-images.githubusercontent.com/51149817/146992747-7d463d20-dc55-4250-bcab-a8433c8f7fdd.png)

**Comment by Josh-Cena** [comment, marked as answer]
Hi, are you using the navbar? Our TOC accesses the navbar's height so if you don't have a navbar there would be an error. Maybe you could give more details about your use-case so we can find a work-around?

**Reply by AngeloCG97** [reply]
Yes, I use navigation bar but I edited it to customize in my project.

Do you know what is the component that calls this method to edit it and prevent it from breaking?

I was checking these two components but I can't find where to call the method that is causing the error
- https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/TOCItems/index.tsx
- https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-classic/src/theme/TOC/index.tsx

**Reply by Josh-Cena** [reply]
Yes, it's here:

https://github.com/facebook/docusaurus/blob/main/packages/docusaurus-theme-common/src/utils/useTOCHighlight.ts#L99

So you have to have an element with class name `.navbar`

**Reply by AngeloCG97** [reply]
Excellent thanks, I can also import these themes into my project and edit them right?

just as I have done with these others.

https://github.com/eoscostarica/eoscostarica.io/tree/master/src/theme

**Reply by Josh-Cena** [reply]
Yes, you can re-implement what's in theme-common if that doesn't fit your need. But I'm still curious why `.navbar` class doesn't exist in your app.

**Reply by AngeloCG97** [reply]
Ok fine, I am going to edit the component to suit my needs.

In this link is the project, it is opensource in case you want to take a look and leave the doubt

https://github.com/eoscostarica/eoscostarica.io

**Reply by Josh-Cena** [reply]
I see, the problem is from here: https://github.com/eoscostarica/eoscostarica.io/blob/master/src/theme/Navbar.js#L233

If `navBar` className is not to be changed, you should add another `navbar` className

**Reply by AngeloCG97** [reply]
So you say that the solution would be to change the name of the navBar class to something else?

**Reply by Josh-Cena** [reply]
Yes, if it won't break anything else, `.navbar` would make the TOC highlighting work

**Reply by AngeloCG97** [reply]
I understand, I'm going to try that solution, to see how it goes.
Thanks a lot.

**Reply by AngeloCG97** [reply]
Thanks, that was the solution
